core/dbt/contracts/graph/manifest.py

```diff
@@ -729,8 +729,12 @@ def patch_nodes(self) -> None:
         # were ok with doing an O(n*m) search (one nodes scan per patch)
         # Q: could we save patches by node unique_ids instead, or convert
         # between names and node ids?
+        patch_keys = set(self.patches.keys())
+        used_patch_keys = set()
+
         for node in self.nodes.values():
-            patch = self.patches.pop(node.name, None)
+            patch_lookup_key = node.patch_lookup_key
+            patch = self.patches.get(patch_lookup_key, None)
             if not patch:
                 continue
```

Comment:
This is new/diverging behavior! Previously, a patch can only be applied once to a single node. The new behavior allows patches to be applied to multiple nodes to support applying the same patch to multiple instances of the same generic test.